src/main/java/com/bravo/user/controller/PaymentController.java

```diff
@@ -0,0 +1,33 @@
+package com.bravo.user.controller;
+
+import com.bravo.user.annotation.SwaggerController;
+import com.bravo.user.model.dto.PaymentDto;
+import com.bravo.user.service.PaymentService;
+import com.bravo.user.validator.UserValidator;
+import lombok.NonNull;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseBody;
+
+import java.util.List;
+
+@RequestMapping(value = "/payment")
+@SwaggerController
+public class PaymentController {
+
+	private final PaymentService paymentService;
+	private final UserValidator userValidator;
+
+	public PaymentController(PaymentService paymentService, UserValidator userValidator) {
+		this.paymentService = paymentService;
+		this.userValidator = userValidator;
+	}
+
+	@GetMapping(value = "/retrieve/{userId}")
+	@ResponseBody
+	public List<PaymentDto> retrieve(@NonNull final @PathVariable String userId) {
```

Comment:
What benefit does `@NonNull` provide here?

Reply:
I added the @NonNull annotation here to make sure that a NullPointerException would be thrown if this method is ever called without userId since userId is required for this method.

Reply:
I just tested this some more by making curl calls to the API on my terminal.  It seems like even if I don't provide a userId the API just treats this as a userId with the value of EMPTYSTRING so I think you are right and this check is not needed.  I think the only way userId will be null is if someone specifically passes in an encoded version of null and when I tested that, it was causing the API to throw a BadRequest exception so I think this is already handled.